tf-psa-crypto/core/psa_util.c

```diff
@@ -365,7 +365,23 @@ int mbedtls_psa_get_random(void *p_rng,
 
 #endif /* MBEDTLS_PSA_CRYPTO_CLIENT */
 
-#if defined(PSA_HAVE_ALG_SOME_ECDSA)
+#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)
```

Comment:
The only places where we're using these macros are guarded by `MBEDTLS_PSA_UTIL_HAVE_ECDSA` which now depends on `PSA_CRYPTO_CLIENT`.

I think there was some confusion about the strategy here: Gilles suggested we guard the functions with `PSA_CRYPTO_CLIENT`, then I suggested that we don't have to do that, we can instead introduce macros with different definitions depending on whether it's enabled... but later I was convinced by Gilles's argument that it's simpler to just remove the functions when PSA's not available. (Said otherwise: the two suggestions were mutually exclusive alternatives.)

Can you remove those macros, and just leave the part when `PSA_UTIL_HAVE_ECDSA` now depends on `PSA_CRYPTO_CLIENT`? (Plus the changes in test dependencies which are good as well.)